File: app/src/main/java/com/karigor/friends/FriendListActivity.java
package com.karigor.friends;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.DefaultItemAnimator;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;

import com.karigor.friends.adapter.FriendListAdapter;
import com.karigor.friends.model.Result;
import com.karigor.friends.viewmodel.FriendListViewModel;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;

public class FriendListActivity extends AppCompatActivity {

    @BindView(R.id.friend_list_view)
    RecyclerView friendListView;

    private FriendListAdapter friendListAdapter;
    private FriendListViewModel viewModel;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ButterKnife.bind(this);

        setupPackageRV();

        getSupportActionBar().setTitle("Friend List");
        viewModel = new ViewModelProvider(this).get(FriendListViewModel.class);
        viewModel.init(this);
        viewModel.getResults().observe(this, new Observer<List<Result>>() {
            @Override
            public void onChanged(List<Result> results) {
                if (results != null) {
                    friendListAdapter.setResults(results);
                }
            }
        });

//        viewModel.CallApiForGetFriends(this);

    }

    private void setupPackageRV() {
        GridLayoutManager gridLayoutManager = new GridLayoutManager(getApplicationContext(), 2);
        friendListView.setLayoutManager(gridLayoutManager);
        friendListView.setItemAnimator(new DefaultItemAnimator());
        friendListAdapter = new FriendListAdapter(this);
        friendListView.setAdapter(friendListAdapter);

    }
}

File: app/src/main/java/com/karigor/friends/FriendListActivity.kt
package com.karigor.friends

import androidx.appcompat.app.AppCompatActivity
import butterknife.BindView
import com.karigor.friends.R
import androidx.recyclerview.widget.RecyclerView
import com.karigor.friends.adapter.FriendListAdapter
import com.karigor.friends.viewmodel.FriendListViewModel
import android.os.Bundle
import androidx.lifecycle.Observer
import butterknife.ButterKnife
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.DefaultItemAnimator
import com.karigor.friends.model.Result

class FriendListActivity : AppCompatActivity() {

    @BindView(R.id.friend_list_view) lateinit var friendListView: RecyclerView
    private var friendListAdapter: FriendListAdapter? = null
    private var viewModel: FriendListViewModel? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        supportActionBar!!.title = "Friend List"

        ButterKnife.bind(this)
        setupPackageRV()

        viewModel = ViewModelProvider(this).get(FriendListViewModel::class.java)
        viewModel!!.init(this)
        viewModel!!.results?.observe(this,
            { results ->
                if (results != null) {
                    friendListAdapter!!.setResults(results)
                }
            })

//        viewModel.CallApiForGetFriends(this);
    }

    private fun setupPackageRV() {
        val gridLayoutManager = GridLayoutManager(applicationContext, 2)
        friendListView!!.layoutManager = gridLayoutManager
        friendListView!!.itemAnimator = DefaultItemAnimator()
        friendListAdapter = FriendListAdapter(this)
        friendListView!!.adapter = friendListAdapter
    }
}